go/vt/vtgate/plan_execute.go

```diff
@@ -29,13 +29,26 @@ import (
 	"vitess.io/vitess/go/vt/vtgate/planbuilder"
 )
 
-func (e *Executor) newExecute(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, logStats *LogStats) (sqlparser.StatementType, *sqltypes.Result, error) {
+func (e *Executor) newExecute(
+	ctx context.Context,
+	safeSession *SafeSession,
+	sql string,
+	bindVars map[string]*querypb.BindVariable,
+	logStats *LogStats,
+	execPlan func(
+		plan *engine.Plan,
+		vc *vcursorImpl,
+		bindVars map[string]*querypb.BindVariable,
+		time time.Time,
+	) error,
+	recResult func(stmtType sqlparser.StatementType, result *sqltypes.Result),
```

Comment:
we should type `execPlan` and `recResult` functions

Reply:
> we should type execPlan and recResult functions

I don't follow. what do you mean?

Reply:
I meant, create two types for those two functions to simplify the code

i.e:

```go
type execPlanFunc func(*engine.Plan, *vcursorImpl, map[string]*querypb.BindVariable, time.Time) error
```